src/rpc/rawtransaction.cpp

```diff
@@ -158,10 +158,12 @@ UniValue getrawtransaction(const JSONRPCRequest& request)
 
     CTransactionRef tx;
     uint256 hashBlock;
-    if (!GetTransaction(hash, tx, Params().GetConsensus(), hashBlock, true))
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, std::string(fTxIndex ? "No such mempool or blockchain transaction"
-            : "No such mempool transaction. Use -txindex to enable blockchain transaction queries") +
-            ". Use gettransaction for wallet transactions.");
+    if (!GetTransaction(hash, tx, Params().GetConsensus(), hashBlock, true)) {
+        std::string msg = std::string(fTxIndex ? "No such mempool or blockchain transaction." :
+                                                 "No such mempool transaction. Use -txindex to enable blockchain transaction queries. "
+                                                 "Use gettransaction for wallet transactions, if wallet is enabled.");
```

Comment:
*for wallet transactions* hence wallet must enabled, therefore *, if wallet is enabled* not needed?